packages/discord.js/src/structures/Message.js

```diff
@@ -30,127 +30,401 @@ const Util = require('../util/Util');
+   * Only available if the message comes from a guild where the author is still a member
+   * @type {?GuildMember}
+   * @readonly
+   */
+  get member() {
+    return this.guild?.members.resolve(this.author) ?? null;
+  }
+
+  /**
+   * Flags that are applied to the message
+   * @type {Readonly<MessageFlagsBitField>}
+   */
+  get flags() {
+    if (!this.data.flags) return null;
+    return new MessageFlagsBitField(this.data.flags);
+  }
+
+  /**
+   * Message reference data
+   * @type {?MessageReference}
+   */
+  get reference() {
+    if (!this.data.message_reference) return null;
+    return {
+      channelId: this.data.message_reference.channel_id,
+      guildId: this.data.message_reference.guild_id,
+      messageId: this.data.message_reference.message_id,
+    };
+  }
+  /**
+   * Partial data of the interaction that this message is a reply to
+   * @type {?MessageInteraction}
+   */
+  get interaction() {
+    if (!this.data.interaction) return null;
+    return {
+      id: this.data.interaction.id,
+      type: this.data.interaction.type,
+      commandName: this.data.interaction.name,
+      user: this.client.users.get(this.data.interaction.user),
+    };
+  }
+
+  /**
+   * The channel that the message was sent in
+   * @type {TextChannel|DMChannel|NewsChannel|ThreadChannel}
+   * @readonly
+   */
+  get channel() {
+    return this.client.channels.resolve(this.channelId);
+  }
+
+  /**
+   * Whether or not this message is a partial
+   * @type {boolean}
+   * @readonly
+   */
+  get partial() {
+    return typeof this.content !== 'string' || !this.author;
+  }
+
+  /**
+   * The time the message was sent at
+   * @type {Date}
+   * @readonly
+   */
+  get createdAt() {
+    return new Date(this.createdTimestamp);
+  }
+
+  /**
+   * The time the message was last edited at (if applicable)
+   * @type {?Date}
+   * @readonly
+   */
+  get editedAt() {
+    return this.editedTimestamp && new Date(this.editedTimestamp);
+  }
+
+  /**
+   * The guild the message was sent in (if in a guild channel)
+   * @type {?Guild}
+   * @readonly
+   */
+  get guild() {
+    return this.client.guilds.resolve(this.guildId) ?? this.channel?.guild ?? null;
+  }
+
+  /**
+   * Whether this message has a thread associated with it
+   * @type {boolean}
+   * @readonly
+   */
+  get hasThread() {
+    return this.flags.has(MessageFlags.HasThread);
+  }
+
+  /**
+   * The URL to jump to this message
+   * @type {string}
+   * @readonly
+   */
+  get url() {
+    return `https://discord.com/channels/${this.guildId ?? '@me'}/${this.channelId}/${this.id}`;
+  }
+
+  /**
+   * The message contents with all mentions replaced by the equivalent text.
+   * If mentions cannot be resolved to a name, the relevant mention in the message content will not be converted.
+   * @type {?string}
+   * @readonly
+   */
+  get cleanContent() {
+    // eslint-disable-next-line eqeqeq
+    return this.content != null ? Util.cleanContent(this.content, this.channel) : null;
```

Comment:
Why not do this instead? 

```suggestion
    return typeof this.content === 'string' ? Util.cleanContent(this.content, this.channel) : null;
```